src/transformers/models/deberta_v2/configuration_deberta_v2.py

```diff
@@ -139,3 +142,19 @@ def __init__(
         self.pooler_hidden_size = kwargs.get("pooler_hidden_size", hidden_size)
         self.pooler_dropout = pooler_dropout
         self.pooler_hidden_act = pooler_hidden_act
+
+
+class DebertaV2OnnxConfig(OnnxConfig):
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        if self.task == "multiple-choice":
+            dynamic_axis = {0: "batch", 1: "choice", 2: "sequence"}
+        else:
+            dynamic_axis = {0: "batch", 1: "sequence"}
+        return OrderedDict(
+            [("input_ids", dynamic_axis), ("attention_mask", dynamic_axis), ("token_type_ids", dynamic_axis)]
+        )
```

Comment:
```suggestion
        if self._config.type_vocab_size > 0:
            return OrderedDict(
                [
                    ("input_ids", dynamic_axis),
                    ("attention_mask", dynamic_axis),
                    ("token_type_ids", dynamic_axis)
                ]
            )
        else:
            return OrderedDict(
                [
                    ("input_ids", dynamic_axis),
                    ("attention_mask", dynamic_axis),
                ]
            )
```
WDYT

Reply:
I'm trying it out now!

Reply:
If I add a value override like

```python
@property
  def values_override(self) -> Optional[Mapping[str, Any]]:
    if hasattr(self._config, "type_vocab_size"):
       return {"type_vocab_size": 2}
```

Then the actual forward pass fails because the underlying model is still initialized without the token type embeddings like

```bash
FAILED tests/onnx/test_onnx_v2.py::OnnxExportTestCaseV2::test_pytorch_export_043_deberta_v2_default - AttributeError: 'DebertaV2Embeddings' object has no attribute 'token_type_embeddings'
```

Reply:
@JingyaHuang The tests pass again with an overridden `generate_dummy_inputs` where if the type vocab size is 0 we delete `type_token_ids` from the dummy inputs. 

**BE WARNED**:
This means that when someone (like me) is actually using this ONNX graph in production they will need to also cleanse the `type_token_ids` from the output of their tokenizer before passing it to the model. There might also be downstream consequences when I work on the quantization pieces for the optimum library.

Reply:
@sam-h-bean 
For the first concern on deleting the `token_type_ids`, as `onnxruntime.InferenceSession` only accepts valid inputs, if the exported ONNX doesn't take `token_type_ids` as input(the case `type_vocab_size=0`), you can only delete the invalid inputs before sending it to the inference session. 

I find it might be interesting to include a filter in [ORTModel](https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/modeling_ort.py#L317), gently pining @philschmid. Does it make sense to consider this kind of edge case and add a filter into the forward pass to filter out invalid inputs?

Reply:
@JingyaHuang So what does that change look like in practice?

Reply:
For the second concern on the quantization, there is __no impact__ if you are doing __dynamic quantization__. 

If you want to do static quantization and have some worries about the computed range of the calibration dataset. There is __no impact on the performance__, but if the ONNX IR doesn't have `token_type_ids` as input, and there are `token_type_ids` in the calibration dataset it will throw out an error. To avoid this, please use [`get_calibration_dataset`](https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/quantization.py#L375) which clean the calibration dataset by applying [`clean_calibration_dataset`](https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/quantization.py#L375).

P.S. I am going to try a nit on tracing the ONNX graph to see if we can avoid so many changes.

Reply:
OK sounds good! So what else needs to happen on this PR to see an approval?